sync/task_pool.go

```diff
@@ -158,14 +159,54 @@ func (p *TaskPool) run(id int, q chan task) error {
 	}
 }
 
-// add task
+// AddTask wait idle worker add task
 func (p *TaskPool) AddTask(t task) {
+	idx := atomic.AddUint32(&p.idx, 1)
+	id := idx % uint32(p.tQNumber)
+
+	select {
+	case <-p.done:
+		return
+	case p.qArray[id] <- t:
+	}
+}
+
+// AddTaskAlways add task to queues or do it immediately
+func (p *TaskPool) AddTaskAlways(t task) {
```

Comment:
suggest to rename func name to `ImmediateAddTask`

Reply:
how about `AddTaskImmediately`? same prefix is friendly for find and  IDE's auto-complete

Reply:
agree